### PR TITLE
feat: support showing jobs sorted by id

### LINKED ISSUE #3173: show sorted jobs
`openmldb_tool inspect offline ls` 可以获得排序后的jobs列表，不检查具体job，检查其中的failed job仍然用`openmldb_tool inspect offline`

`pip show openmldb-tool`获得`Location: <path>`，替换`<path>/diagnostic_tool/diagnose.py`为新版https://gist.github.com/vagetablechicken/f6f0a6a6ec0a4b18b0547dff21ad3618

- [ ] pr

### PR DESCRIPTION
**Pull request by emo-coder**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behavior?** (You can also link to an open issue here)
close #3173 

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

File: src/sdk/job_table_helper.cc [hunk 1]
```diff
@@ -164,6 +165,10 @@ std::shared_ptr<hybridse::sdk::ResultSet> JobTableHelper::MakeResultSet(
         vec.push_back("TaskManager");
         records.emplace_back(std::move(vec));
     }
+    // sort jobs by id
+    std::sort(records.begin(), records.end(),
+              [](std::vector<std::string> vec1, std::vector<std::string> vec2) {
+              return atoi(vec1[0].c_str()) < atoi(vec2[0].c_str()); });
```
Comment (dl239) [inline src/sdk/job_table_helper.cc]: - use `absl::SimpleAtoi` and tackle the non number case
- judge empty

File: src/sdk/sql_cluster_test.cc [hunk 2]
```diff
@@ -98,6 +99,107 @@ class SQLClusterDDLTest : public SQLClusterTest {
...
+    sql = "insert into " + name +
+          " values(1, \"ImportOnlineData\", \"Finished\", 1000, 1001, "
+          "\"/tmp/sql-00000000000000000001\", \"local[*]\", \"local-000000000000001\", \"\");";
+    router->ExecuteSQL(db, sql, &status);
+    ASSERT_TRUE(status.IsOK());
+
+    sql = "insert into " + name +
+          " values(5, \"ImportOnlineData\", \"Failed\", 1000, 1005, "
+          "\"/tmp/sql-00000000000000000005\", \"local[*]\", \"local-000000000000005\", \"\");";
+    router->ExecuteSQL(db, sql, &status);
```
Comment (dl239) [inline src/sdk/sql_cluster_test.cc]: insert in a loop

File: src/sdk/job_table_helper.cc [hunk 3]
```diff
@@ -164,6 +165,10 @@ std::shared_ptr<hybridse::sdk::ResultSet> JobTableHelper::MakeResultSet(
         vec.push_back("TaskManager");
         records.emplace_back(std::move(vec));
     }
+    // sort jobs by id
+    std::sort(records.begin(), records.end(),
+              [](std::vector<std::string> vec1, std::vector<std::string> vec2) {
```
Comment (dl239) [inline src/sdk/job_table_helper.cc]: use const reference

File: src/sdk/job_table_helper.cc [hunk 4]
```diff
@@ -164,6 +165,21 @@ std::shared_ptr<hybridse::sdk::ResultSet> JobTableHelper::MakeResultSet(
...
+    // sort jobs by id(asc)
+    std::sort(records.begin(), records.end(),
+              [](const std::vector<std::string> vec1, const std::vector<std::string> vec2) {
+              if (vec1.empty()) {
+                  return true;
+              }
+              if (vec2.empty()) {
+                  return false;
+              }
+              int id1, id2;
```
Comment (dl239) [inline src/sdk/job_table_helper.cc]: uint64_t

File: src/sdk/sql_cluster_test.cc [hunk 5]
```diff
@@ -98,6 +100,50 @@ class SQLClusterDDLTest : public SQLClusterTest {
...
+          "("
+          "id int, job_type string, state string, start_time timestamp, end_time timestamp, "
+          "parameter string, cluster string, application_id string, error string, "
+          "index(key=id));";
+    ASSERT_TRUE(router->ExecuteDDL(db, sql, &status)) << "ddl: " << sql;
+    ASSERT_TRUE(router->RefreshCatalog());
+
+    uint32_t seed = time(NULL);
+    std::vector<int> randint;
+    for (int i = 0; i < rand_r(&seed); i++) {
```
Comment (dl239) [inline src/sdk/sql_cluster_test.cc]: keep fixed size

File: src/sdk/sql_cluster_test.cc [hunk 6]
```diff
@@ -27,6 +28,7 @@
 #include "gtest/gtest.h"
 #include "sdk/mini_cluster.h"
 #include "sdk/sql_cluster_router.h"
+#include "sdk/job_table_helper.h"
```
Comment (dl239) [inline src/sdk/sql_cluster_test.cc]: reorder the header file

File: src/sdk/job_table_helper.cc [hunk 7]
```diff
@@ -17,6 +17,7 @@
 #include "sdk/job_table_helper.h"
 
 #include <unordered_map>
+#include <algorithm>
```
Comment (dl239) [inline src/sdk/job_table_helper.cc:19]: reorder the header file

**Review by dl239 — APPROVED** [review]

**Review by vagetablechicken — APPROVED** [review]

**Review by lqy222 — APPROVED** [review]

**Pull request merged**